test/data_type/test_custom_type.cpp

```diff
@@ -0,0 +1,831 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2024, Advanced Micro Devices, Inc. All rights reserved.
+
+#include "gtest/gtest.h"
+#include "ck/utility/data_type.hpp"
+#include "ck/utility/type_convert.hpp"
+
+using ck::bf8_t;
+using ck::bhalf_t;
+using ck::f8_t;
+using ck::half_t;
+using ck::Number;
+using ck::type_convert;
+using ck::vector_type;
+
+TEST(Custom_bool, TestSize)
+{
+    struct custom_bool_t
+    {
+        bool data;
+    };
+    ASSERT_EQ(sizeof(bool), 1);
```

Comment:
I believe, we can only rely on `sizeof(char) == 1` . For the rest of the fundamental types, in general, output of the `sizeof` is implementation defined. 
I propose to verify `sizeof` of derived types relative to the size of the fundamental type without focusing on the specific number. E.g.:
```suggestion
    ASSERT_EQ(sizeof(custom_bool_t), sizeof(bool));
    ASSERT_EQ(sizeof(vector_type<bool, 2>), 2*sizeof(bool));
```

Reply:
Fixed in [76c681b](https://github.com/ROCm/composable_kernel/pull/1333/commits/76c681be7e36e02bcde82f98b4995d2a9c1a6507)